### PR TITLE
Add support for Black's `--pyi` flag

### LINKED ISSUE #374: Support for black's `--pyi` flag?
### Description

Hi! Thanks for maintaining blacken-docs -- it's a great project!

Black supports a `--pyi` flag; if you pass this flag, it formats all code as if it were a `.pyi` stub file. (In practice, this results in more concise formatting with fewer blank lines.) I'm working on a project where we'd ideally like our Python examples in Markdown files formatted quite concisely, as if they were stubs. However, there doesn't seem to be any way currently to do this: black-as-invoked-by-blacken-docs doesn't take notice of black configuration options in a `[tool.black]` pyproject.toml table (I think because you're invoking Black as a library rather than as a subprocess), and blacken-docs has no flag or config option to format docs examples as stubs.

Would you be open to adding a `--pyi` flag to blacken-docs, that would simply be passed directly on to Black? (I'd be very happy to submit a PR, if so!)

### PR DESCRIPTION
**Pull request by AlexWaygood**
Fixes #374

### CODE REVIEW

File: src/blacken_docs/__init__.py [hunk 1]
```diff
@@ -338,14 +338,16 @@ def main(argv: Sequence[str] | None = None) -> int:
         "--rst-literal-blocks",
         action="store_true",
     )
+    parser.add_argument("--pyi", action="store_true")
     parser.add_argument("filenames", nargs="*")
     args = parser.parse_args(argv)
 
-    black_mode = black.FileMode(
+    black_mode = black.Mode(
```
Comment (AlexWaygood) [inline src/blacken_docs/__init__.py:345]: Just a drive-by change, since I spotted that this is just an alias for backwards compatibility in Black these days: https://github.com/psf/black/blob/c98fc0c128e1a79a776b62046acaa28685a31089/src/black/__init__.py#L113-L114

Comment (adamchainz) [reply src/blacken_docs/__init__.py:345]: Sure, thanks!

File: src/blacken_docs/__init__.py [hunk 2]
```diff
@@ -338,14 +338,16 @@ def main(argv: Sequence[str] | None = None) -> int:
...
         line_length=args.line_length,
         preview=args.preview,
         string_normalization=not args.skip_string_normalization,
         target_versions=set(args.target_versions),
+        is_pyi=args.pyi,
```
Comment (adamchainz) [inline src/blacken_docs/__init__.py:350]: Whilst we're at this, let's sort the args to match the definition

```suggestion
        target_versions=set(args.target_versions),
        line_length=args.line_length,
        string_normalization=not args.skip_string_normalization,
        is_pyi=args.pyi,
        preview=args.preview,
```

**Review by adamchainz — APPROVED** [review]

**Comment by AlexWaygood** [comment]
Thanks! :D
